asset-transfer-events/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,166 @@
+
+// Path to peer tls certificate.
+const tlsCertPath = path.resolve(cryptoPath, 'peers', 'peer0.org1.example.com', 'tls', 'ca.crt');
+
+// Gateway peer endpoint.
+const peerEndpoint = 'localhost:7051';
+
+const utf8Decoder = new TextDecoder();
+const now = Date.now();
+const assetId = `asset${now}`;
+
+async function main(): Promise<void> {
+    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
+    const client = await newGrpcConnection();
+
+    const gateway = connect({
+        client,
+        identity: await newIdentity(),
+        signer: await newSigner(),
+    });
+    let events: CloseableAsyncIterable<ChaincodeEvent> | undefined;
+
+    try {
+
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const network = gateway.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contract = network.getContract(chaincodeName);
+
+        //Start Listening to events.
+        events = await startEventListening(network)
+
+        // Create a new asset on the ledger.
+        await createAsset(contract);
+
+        // Update an existing asset asynchronously.
+        await transferAssetAsync(contract);
+
+        // Delete asset by assetID.
+        await deleteAssetByID(contract);
+
+    } finally {
+        gateway.close();
+        client.close();
+        events?.close()
+    }
+}
+
+main().catch(error => console.error('******** FAILED to run the application:', error));
+
+async function newGrpcConnection(): Promise<grpc.Client> {
+    const tlsRootCert = await fs.readFile(tlsCertPath);
+    const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
+    return new grpc.Client(peerEndpoint, tlsCredentials, {
+        'grpc.ssl_target_name_override': 'peer0.org1.example.com',
+    });
+}
+
+async function newIdentity(): Promise<Identity> {
+    const credentials = await fs.readFile(certPath);
+    return { mspId, credentials };
+}
+
+async function newSigner(): Promise<Signer> {
+    const files = await fs.readdir(keyDirectoryPath);
+    const keyPath = path.resolve(keyDirectoryPath, files[0]);
+    const privateKeyPem = await fs.readFile(keyPath);
+    const privateKey = crypto.createPrivateKey(privateKeyPem);
+    return signers.newPrivateKeySigner(privateKey);
+}
+
+async function startEventListening(network: Network) {
+    console.log('Read chaincode events');
+    const events = await network.getChaincodeEvents(chaincodeName);
+    readEvents(events);
+    return events;
+}
+
+async function readEvents(events: CloseableAsyncIterable<ChaincodeEvent>): Promise<void> {
+    try {
+        for await (const event of events) {
+            const payload = utf8Decoder.decode(event.payload);
+            console.log(`Received event name: ${event.eventName}, payload: ${payload}, txID: ${event.transactionId}, blockNumber:${event.blockNumber}`);
```

Comment:
The README says event receipt messages will begin with `<--`, which mirrors the transaction submit messages that begin with `-->`. This message should change to match that pattern.

> the events received at application end (console output like `<-- Contract Event Received:` or `<-- Block Event Received`)